src/scaffoldmaker/meshtypes/meshtype_3d_lung1.py

```diff
@@ -780,3 +684,496 @@ def refineMesh(cls, meshrefinement, options):
+                    # remap parameters on xi3 = 1 before collapsing nodes
+                    remapEftNodeValueLabel(eft, nodes, Node.VALUE_LABEL_D_DS1, [])
+                    remapEftNodeValueLabel(eft, nodes, Node.VALUE_LABEL_D_DS2, [])
+                    remapEftNodeValueLabel(eft, [7, 8], Node.VALUE_LABEL_D_DS3, [(Node.VALUE_LABEL_D_DS2, [1])])
+                    remapEftNodeValueLabel(eft, [6], Node.VALUE_LABEL_D_DS3, [(Node.VALUE_LABEL_D_DS1, [1])])
+                    remapEftNodeValueLabel(eft, [3, 4], Node.VALUE_LABEL_D_DS1, [])
+                    remapEftNodeValueLabel(eft, [4], Node.VALUE_LABEL_D_DS2,
+                                           [(Node.VALUE_LABEL_D_DS1, [1]), (Node.VALUE_LABEL_D_DS2, [])])
+                    ln_map = [1, 2, 3, 3, 4, 4, 4, 4]
+                    remapEftLocalNodes(eft, 4, ln_map)
+
+                elif (e3 == (uElementsCount3 - 2)) and (e2 == (uElementsCount2 - 3)):
+                    # Remapped cube element 1
+                    eft = eftfactory.createEftBasic()
+                    setEftScaleFactorIds(eft, [1], [])
+                    remapEftNodeValueLabel(eft, [3, 4], Node.VALUE_LABEL_D_DS2, [(Node.VALUE_LABEL_D_DS3, [1])])
+                    remapEftNodeValueLabel(eft, [3, 4], Node.VALUE_LABEL_D_DS3,
+                                           [(Node.VALUE_LABEL_D_DS2, []), (Node.VALUE_LABEL_D_DS3, [])])
+                elif (e3 == (uElementsCount3 - 2)) and (e2 == (uElementsCount2 - 2)):
+                    # Remapped cube element 2
+                    eft = eftfactory.createEftBasic()
+                    setEftScaleFactorIds(eft, [1], [])
+                    remapEftNodeValueLabel(eft, [1, 2], Node.VALUE_LABEL_D_DS3,
+                                           [(Node.VALUE_LABEL_D_DS2, []), (Node.VALUE_LABEL_D_DS3, [])])
+                elif None in nodeIdentifiers:
+                    continue
+
+                if eft is eftRegular:
+                    element = mesh.createElement(elementIdentifier, elementtemplateRegular)
+                else:
+                    elementtemplateCustom.defineField(coordinates, -1, eft)
+                    element = mesh.createElement(elementIdentifier, elementtemplateCustom)
+                element.setNodesByIdentifier(eft, nodeIdentifiers)
+                if eft.getNumberOfLocalScaleFactors() == 1:
+                    element.setScaleFactors(eft, [-1.0])
+                elementIdentifier += 1
+                lungMeshGroup.addElement(element)
+                if lungSide == leftLung:
+                    lungSideMeshGroup.addElement(element)
+                    upperLungMeshGroup.addElement(element)
+                else:
+                    lungSideMeshGroup.addElement(element)
+                    if e3 < (uElementsCount3 - 2):
+                        middleRightLungMeshGroup.addElement(element)
+                    else:
+                        upperLungMeshGroup.addElement(element)
```

Comment:
```
lungSideMeshGroup.addElement(element)
if middleLobeMeshGroup:
    middleLobeMeshGroup.addElement(element)
if upperLobeMeshGroup:
    upperLobeMeshGroup.addElement(element)
````

Reply:
I don't think the above will be able to classify middle and upper lobes. In this case, I can change them as shown below.

`                if upperLobeMeshGroup and middleLobeMeshGroup:
                    if e3 < (uElementsCount3 - 2):
                        middleLobeMeshGroup.addElement(element)
                    else:
                        upperLobeMeshGroup.addElement(element) `

Reply:
Yes I'd forgotten about checking which element. This may be better:
```
if middleLobeMeshGroup and (e3 < (uElementsCount3 - 2):
    middleLobeMeshGroup.addElement(element)
elif upperLobeMeshGroup:
    upperLobeMeshGroup.addElement(element)
```